### PR TITLE
Various fixes

### PR DESCRIPTION
**Pull request by h-2**
~~blocked by seqan/seqan#2418~~

### CODE REVIEW

**Comment by sarahet** [comment]
Is this now still not comparable to lambda2 SIMD?

**Comment by h-2** [comment]
Just FYI, they broke some CMAKE stuff in SeqAn master, that's why our CI is failing.

File: src/search_datastructures.hpp [hunk 1]
```diff
@@ -408,11 +408,6 @@ class GlobalDataHolder
     using TQryIds       = TIds;
     using TSubjIds      = TIds;
 
-    /* indeces and their type */
-    using TDbIndex      = std::conditional_t<c_dbIndexType == DbIndexType::BI_FM_INDEX,
```
Comment (sarahet) [inline src/search_datastructures.hpp]: Why don't we need this anymore? I thought we will include the bidirectional index again at some point.

File: src/search_algo.hpp [hunk 2]
```diff
@@ -1166,9 +1165,9 @@ iterateMatchesFullSerial(TLocalHolder & lH)
         bm._n_qId = it->qryId / seqan::qNumFrames(TGlobalHolder::blastProgram);
         bm._n_sId = it->subjId / seqan::sNumFrames(TGlobalHolder::blastProgram);
 
-        bm.sLength = seqan::sIsTranslated(TGlobalHolder::blastProgram)
-                        ? lH.gH.indexFile.seqs[it->subjId].size()
-                        : lH.gH.transSbjSeqs[it->subjId].size();
+        bm.qLength = //std::ranges::size(lH.gH.transQrySeqs[it->qryId]);
```
Comment (sarahet) [inline src/search_algo.hpp]: What about the comment? Is this needed in the future?

File: src/search_datastructures.hpp (hunk 1 above)
Comment (h-2) [reply src/search_datastructures.hpp]: This alias is just a leftover of lambda2, it's independent of everything else (the index type is now part of the index file that we use as-is).

File: src/search_algo.hpp (hunk 2 above)
Comment (h-2) [reply src/search_algo.hpp]: We should discuss this again. Like I said, I think basing the variable on the actual translated seq length would be better for the e-value calculation but for now the focus was on reproducing lambda2's results.

**Pull request merged**